service/service.cc

```diff
@@ -233,6 +243,70 @@ Status ServiceImpl::monitor(ServerContext *context,
   return Status::OK;
 }
 
+// Watching all chirps in key: "this_is_where_i_store_all_the_chirps"
+// const MonitorRequst *request contains the hashtag we want to monitor
+// once received new chirp, check if it has the requested hashtag
+// if it does, send to MonitorReply* reply
+// loop times represent the number of loops monitor want to use
+// default value of loop times is -1 represents loop forever
+// Class private variable std::mutex monitor_mutex_
+// std::condition_variable monitor_buf_signal_
+// bool monitor_flag_ and end_flag_ are used to synchronize monitor
+// and its buffer function MonitorBuffer.
+// if you want to buffer it, remeber to set buff_mode_ to be true
+// by using OpenBuffer() function
+// if you do not want to buffer them, use CloseBuffer() to close
+Status ServiceImpl::stream(ServerContext *context,
+                            const MonitorRequest *request,
+                            ServerWriter<MonitorReply> *reply) {
+  auto time_interval = refresh_timeval_;  // pick one refresh frequency
```

Comment:
Minor: maybe add some newlines to this code to break it up visually into logical blocks.